CorpusCallosum/fastsurfer_cc.py

```diff
@@ -0,0 +1,880 @@
+        default=None,
+    )
+    parser.add_argument(
+        "--softlabels_cc_path",
+        type=str,
+        help=f"Path for cc softlabels (default: subject_dir/{STANDARD_OUTPUT_PATHS['softlabels_cc']})",
+        default=None,
+    )
+    parser.add_argument(
+        "--softlabels_fn_path",
+        type=str,
+        help=f"Path for fornix softlabels (default: subject_dir/{STANDARD_OUTPUT_PATHS['softlabels_fn']})",
+        default=None,
+    )
+    parser.add_argument(
+        "--softlabels_background_path",
+        type=str,
+        help=f"Path for background softlabels (default: subject_dir/{STANDARD_OUTPUT_PATHS['softlabels_background']})",
+        default=None,
+    )
+    parser.add_argument("--verbose", action="store_true", help="Enable verbose (shows output paths)", default=False)
+
+
+    args = parser.parse_args()
+
+    # Validation logic: either subject_dir OR both in_mri and aseg must be provided
+    if not args.subject_dir and (not args.in_mri or not args.aseg):
+        parser.error("You must specify either --subject_dir OR both --in_mri and --aseg arguments.")
+
+    # If subject_dir is provided, set default paths for missing arguments
+    if args.subject_dir:
+        subject_dir_path = Path(args.subject_dir)
+
+        # Create standard FreeSurfer subdirectories
+        (subject_dir_path / "mri").mkdir(parents=True, exist_ok=True)
+        (subject_dir_path / "stats").mkdir(parents=True, exist_ok=True)
+        (subject_dir_path / "transforms").mkdir(parents=True, exist_ok=True)
+
+        if not args.in_mri:
+            args.in_mri = str(subject_dir_path / "mri" / "orig.mgz")
+
+        if not args.aseg:
+            args.aseg = str(subject_dir_path / "mri" / "aparc.DKTatlas+aseg.deep.mgz")
+
+        # Set default output paths if not provided
+        for key, value in STANDARD_OUTPUT_PATHS.items():
+            if not getattr(args, f"{key}_path") and value is not None:
+                setattr(args, f"{key}_path", str(subject_dir_path / value))
+
+        # Set output_dir to subject_dir
+        args.output_dir = str(subject_dir_path)
+
+    # Create parent directories for all output paths
+    for path_name in STANDARD_OUTPUT_PATHS.keys():
+        path = getattr(args, f"{path_name}_path")
+        if path is not None:
+            Path(path).parent.mkdir(parents=False, exist_ok=True)
+
+    return args
+
+
+def centroid_registration(aseg_nib: nib.Nifti1Image, verbose: bool = False) -> tuple[
+    np.ndarray, np.ndarray, np.ndarray, nib.Nifti1Header, np.ndarray
+]:
+    """Perform centroid-based registration between subject and fsaverage space.
+
+    Computes a rigid transformation between the subject's segmentation and fsaverage space
+    by aligning centroids of corresponding anatomical structures.
+
+    Parameters
+    ----------
+    aseg_nib : nibabel.Nifti1Image
+        Subject's segmentation image.
+    verbose : bool, optional
+        Whether to print progress information, by default False.
```

Comment:
```suggestion
def centroid_registration(aseg_nib: nib.Nifti1Image) -> tuple[
    npt.NDArray[float], npt.NDArray[float], npt.NDArray[float], nib.Nifti1Header, npt.NDArray[float]
]:
    """Perform centroid-based registration between subject and fsaverage space.

    Computes a rigid transformation between the subject's segmentation and fsaverage space
    by aligning centroids of corresponding anatomical structures.

    Parameters
    ----------
    aseg_nib : nibabel.Nifti1Image
        Subject's segmentation image.
```

Ideally, the verbosity setting is handled by the logger, i.e. the Logger's output setting.